### PR TITLE
[Merged by Bors] - Deflake failing unit and mining system tests

### LINKED ISSUE #2735: Add nodes system test failure - too many blocks created
## Description
When running add nodes test, sometimes it fails with following error message: 
 `blocks created per layer 1.0, wanted average block per node 0.6666666666666666, layer_avg_size 50, layers_per_epoch 3, num_miners 51, blocks_sum 3, to_layer 18, from_layer 15`

deployment example `acu99` 

## Steps to reproduce
run add_nodes system test

## Environment
Please complete the following information:
 - OS: 
 
 - Node Version: 
 
 - Smapp version (if applicable):

### LINKED ISSUE #2737: panic in github.com/spacemeshos/go-spacemesh/cmd/node
## Description
```
panic: Log in goroutine after TestSpacemeshApp_P2PInterface has completed

goroutine 870 [running]:
testing.(*common).logDepth(0xc0090d0300, 0xc0000f56c0, 0xdc, 0x3)
	/opt/hostedtoolcache/go/1.15.13/x64/src/testing/testing.go:755 +0x5df
testing.(*common).log(...)
	/opt/hostedtoolcache/go/1.15.13/x64/src/testing/testing.go:737
testing.(*common).Logf(0xc0090d0300, 0x118766d, 0x2, 0xc00c77d990, 0x1, 0x1)
	/opt/hostedtoolcache/go/1.15.13/x64/src/testing/testing.go:783 +0x7e
go.uber.org/zap/zaptest.testingWriter.Write(0x7f7427e12d78, 0xc0090d0300, 0x203000, 0xc00cde0000, 0xdd, 0x400, 0xc0000c8e20, 0xfc4980, 0xc00c77d980)
	/home/runner/go/pkg/mod/go.uber.org/zap@v1.15.0/zaptest/logger.go:130 +0x123
go.uber.org/zap/zapcore.(*ioCore).Write(0xc00cda7470, 0x0, 0xc042a0ce28f1b77c, 0x3966592f9, 0x1a6f240, 0x0, 0x0, 0x11ae3ec, 0x22, 0x0, ...)
	/home/runner/go/pkg/mod/go.uber.org/zap@v1.15.0/zapcore/core.go:90 +0x106
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc00c736790, 0xc00cd70f00, 0x2, 0x2)
	/home/runner/go/pkg/mod/go.uber.org/zap@v1.15.0/zapcore/entry.go:216 +0x12d
go.uber.org/zap.(*Logger).Info(0xc00cda8f60, 0x11ae3ec, 0x22, 0xc00cd70f00, 0x2, 0x2)
	/home/runner/go/pkg/mod/go.uber.org/zap@v1.15.0/logger.go:187 +0x85
github.com/spacemeshos/go-spacemesh/log.FieldLogger.Info(0xc00cda8f60, 0x0, 0x0, 0x11ae3ec, 0x22, 0xc004ea5a10, 0x1, 0x1)
	/home/runner/work/go-spacemesh/go-spacemesh/log/zap.go:229 +0x1d0
github.com/spacemeshos/go-spacemesh/p2p/gossip.(*Protocol).handlePQ(0xc00ce176c0, 0x130f820, 0xc00d0375c0)
	/home/runner/work/go-spacemesh/go-spacemesh/p2p/gossip/protocol.go:157 +0x119f
created by github.com/spacemeshos/go-spacemesh/p2p/gossip.(*Protocol).propagationEventLoop
	/home/runner/work/go-spacemesh/go-spacemesh/p2p/gossip/protocol.go:211 +0x71
FAIL	github.com/spacemeshos/go-spacemesh/cmd/node	15.417s
```
## Steps to reproduce
run unit tests 

## Environment
Please complete the following information:
 - OS: 
 
 - Node Version: 
 
 - Smapp version (if applicable):

### LINKED ISSUE #2738: unit tests fail in TestFindNodeProtocol_FindNode_Concurrency
## Description
FAIL: TestFindNodeProtocol_FindNode_Concurrency

https://github.com/spacemeshos/go-spacemesh/runs/3438363940

## Environment
Please complete the following information:
 - OS: 
 
 - Node Version: 
 
 - Smapp version (if applicable):

### PR DESCRIPTION
**Pull request by dshulyak**
## Motivation

closes: https://github.com/spacemeshos/go-spacemesh/issues/2738
closes: https://github.com/spacemeshos/go-spacemesh/issues/2737
closes: https://github.com/spacemeshos/go-spacemesh/issues/2735
And TestBlockBuilder_notSynced

## TODO

- [ ] Explain motivation or link existing issue(s)
- [ ] Test changes and document test plan
- [ ] Update documentation as needed

## DevOps Notes

- [x] This PR does not require configuration changes (e.g., environment variables, GitHub secrets, VM resources)
- [x] This PR does not affect public APIs
- [x] This PR does not rely on a new version of external services (PoET, elasticsearch, etc.)
- [x] This PR does not make changes to log messages (which monitoring infrastructure may rely on)

### CODE REVIEW

**Comment by dshulyak** [comment]
bors try

**Comment by dshulyak** [comment]
bors try

**Comment by dshulyak** [comment]
bors try

**Comment by dshulyak** [comment]
bors try

File: p2p/gossip/protocol.go [hunk 1]
```diff
@@ -208,8 +230,6 @@ func (p *Protocol) isShuttingDown() bool {
 
 // pushes messages that passed validation into the priority queue
 func (p *Protocol) propagationEventLoop(ctx context.Context) {
-	go p.handlePQ(ctx)
```
Comment (nkryuchkov) [inline p2p/gossip/protocol.go:211]: Why should this be removed?

Comment (dshulyak) [reply p2p/gossip/protocol.go:211]: i moved it. please see full change

**Review by nkryuchkov — APPROVED** [review]

**Comment by dshulyak** [comment]
bors try

**Comment by dshulyak** [comment]
bors try

**Comment by dshulyak** [comment]
bors try

**Comment by dshulyak** [comment]
Just a note, raising hash to 32 bytes eliminated those dropped messages. We still have another issues, but atx seems to be eventually delivered everywhere.

**Comment by dshulyak** [comment]
bors try

**Comment by dshulyak** [comment]
bors try

**Comment by dshulyak** [comment]
bors try

**Comment by dshulyak** [comment]
bors try

**Comment by dshulyak** [comment]
p2p failed.

bors try

**Comment by dshulyak** [comment]
bors try

**Comment by dshulyak** [comment]
bors try

**Comment by dshulyak** [comment]
bors try

**Comment by dshulyak** [comment]
bors try

**Comment by dshulyak** [comment]
bors try

**Comment by dshulyak** [comment]
bors try

**Comment by dshulyak** [comment]
bors merge

**Pull request closed without merging**